### PR TITLE
Copter: add AUTO RTL pseudo mode, allowing DO_LAND_START

### PR DESCRIPTION
**Pull request by IamPete1**
This is now a pseudo mode, AUTO will report on as this mode so long a mission still thinks its in a landing sequence. We get the best of being a mode, GCS mode switch and reporting. If you change the wp index you will 'change' from Auto RTL back to auto.

Quite pleased with this now, I don't think there is much missing. If you were in Auto RTL 'mode' and change to auto you will just carry on to the same waypoint. We don't restore the auto state.

### CODE REVIEW

File: ArduCopter/events.cpp [hunk 1]
```diff
@@ -170,6 +173,9 @@ void Copter::failsafe_gcs_on_event(void)
         case FS_GCS_ENABLED_ALWAYS_LAND:
             desired_action = Failsafe_Action_Land;
             break;
+        case FS_THR_ENABLED_AUTO_RTL_OR_RTL:
+            desired_action = Failsafe_Action_Auto_DO_lAND_START;
```
Comment (rmackay9) [inline ArduCopter/events.cpp:177]: small typo in that it has a small "l".

File: ArduCopter/Parameters.cpp [hunk 2]
```diff
@@ -1089,6 +1089,13 @@ const AP_Param::GroupInfo ParametersG2::var_info[] = {
     AP_GROUPINFO("TERRAIN_MARGIN", 47, ParametersG2, terrain_margin, 10.0),
 #endif
 
+    // @Param: AUTO_RTL
```
Comment (rmackay9) [inline ArduCopter/Parameters.cpp:1092]: I think we can skip this parameter can't we?  That would be nice 'cuz I'm not really a fan of the name.  It's not terrible but it is all on it's own and is the only parameter starting with "AUTO_".   Not a blocker but if possible I'd like to do without this parameter for now.

**Comment by rmackay9** [comment]
This looks pretty good to me..

File: ArduCopter/Parameters.cpp (hunk 2 above)
Comment (IamPete1) [reply ArduCopter/Parameters.cpp:1092]: Yeah, the param is pointless in this PR, I have 3 other methods to add so will need it, but that will be a second PR.

**Comment by rmackay9** [comment]
Merged, thanks!

**Pull request merged**